aws-crt/spec/event_loop_group_spec.rb

```diff
@@ -1,8 +1,19 @@
 require 'spec_helper'
+require 'weakref'
 
-describe Aws::Crt::IO do
-  it 'has a test' do
+describe Aws::Crt::IO::EventLoopGroup do
+  it 'lives and dies' do
+    # create
     elg = Aws::Crt::IO::EventLoopGroup.new
-    elg.destroy
+    expect(elg).to_not be_nil
+    weakref = WeakRef.new(elg)
+    expect(weakref.weakref_alive?).to be_truthy
+
+    # destroy
+    elg = nil # rubocop:disable Lint/UselessAssignment
+    ObjectSpace.garbage_collect
+    expect(weakref.weakref_alive?).to be_falsey
+
+    check_for_clean_shutdown
```

Comment:
When do we need to call this?  Will we need to call this after every test, after the full test run.... ect?  Or is this something that we would just selectively call during specific tests?

Reply:
We'll want to call this at the end of every test that creates CRT objects.
Is there an easy way to get this to run automatically at the end of each test?

Reply:
Theres a few ways we can get that depending on how exactly we want to set this up:

1. In each spec file (or even in specific context/describe blocks) that is creating CRT objects add: `after(:each) { check_for_clean_shutdown}` 
2. Add it to the global config so that it is called after every test.  In the spec_helper.rb  we'd add something like:
```ruby
RSpec.configure do |config|
  config.after(:each) { check_for_clean_shutdown }
end
```
3.  Use a custom Rspec "type" to define the after for only appropriate test suits.  Eg:
```ruby
describe Aws::Crt::IO::EventLoopGroup, type: :crt do
# spec code
end

# spec helper
RSpec.configure do |config|
  config.after(:each, type: :crt) { check_for_clean_shutdown }
end
```